lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -3133,18 +3138,25 @@ namespace Js
                     converted = CopyNativeIntArrayElementsToFloat(pDestArray, idxDest, pIntArray);
                     idxDest = idxDest + pIntArray->length;
                 }
-                else
+                else if (JavascriptNativeFloatArray::Is(aItem))
                 {
                     JavascriptNativeFloatArray* pItemArray = JavascriptNativeFloatArray::FromVar(aItem);
                     converted = CopyNativeFloatArrayElements(pDestArray, idxDest, pItemArray);
                     idxDest = idxDest + pItemArray->length;
                 }
+                else
```

Comment:
With -es6isconcatspreadable on, JavascriptOperators::IsConcatSpreadable() call may run user code that potentially alters the array type. Search "getter altering array type" in es6IsConcatSpreadable.js

---
In reply to: [68432695](https://github.com/Microsoft/ChakraCore/pull/1195#discussion_r68432695) [](ancestors = 68432695)